src/core/Akka.TestKit/TestKitBase_AwaitAssert.cs

```diff
@@ -59,9 +62,15 @@ public async Task AwaitAssertAsync(Action assertion, TimeSpan? duration=null, Ti
                 }
                 catch(Exception)
                 {
-                    if(Now + t >= stop)
+                    var stopped = Now + t;
+                    if (stopped >= stop)
+                    {
+                        Sys.Log.Warning("AwaitAssert failed, timeout [{0}] is over after [{1}] attempts and [{2}] elapsed time", max, attempts, stopped - start);
```

Comment:
Diagnostic improvements to `AwaitAssert` - I wanted to understand how many times an assertion actually ran before it failed, so I capture that data and the total elapsed time the test used and pipe it into a `Warning` log right before we `throw` the assertion exception.